### PR TITLE
UMat fixes (getUMat & 'map_unmap_counting')

### PR DESCRIPTION
**Pull request by apavlenko**
fixes for the following failing UMat tests: `Core_UMat.getUMat` (#5010), `UMat.map_unmap_counting` (#5259)

### CODE REVIEW

File: modules/core/src/ocl.cpp [hunk 1]
```diff
@@ -4572,16 +4572,13 @@ class OpenCLAllocator : public MatAllocator
                     else
                     {
                         // CL_MEM_USE_HOST_PTR (nothing is required) and OTHER cases
```
Comment (alalek) [inline modules/core/src/ocl.cpp]: This comment is not actual, please remove it.

**Comment by vpisarev** [comment]
so, we decided that we need a solution that works well on both Intel and AMD, even with temporarily broken drivers.

**Comment by apavlenko** [comment]
Vadim, do you consider removing `UMatData`? :)

**Comment by vpisarev** [comment]
@apavlenko, emm, no, why would I? :)

**Comment by alalek** [comment]
:+1:

**Pull request merged**